core/commands/add.go

```diff
@@ -276,6 +276,7 @@ You can now check what blocks have been created by:
 		fileAdder.NoCopy = nocopy
 		fileAdder.Name = pathName
 		fileAdder.CidBuilder = prefix
+		fileAdder.Hash = hash
```

Comment:
Shouldn't this just set `fileAdder.Pin` to false?

Reply:
why should put `fileAdder.Pin` to false? The reason i add `fileAdder.Hash` is that i want to get this param in `coreunix/add.go`.

Reply:
I know, but we don't really need this param there as all it does in disable pinning. See https://github.com/ipfs/go-ipfs/blob/b789295580d2ede0df6e248d73a1a7fb60b253ba/core/coreunix/add.go#L171-L178

We should just do `fileAdder.Pin = dopin && hash`

Reply:
And maybe we should do `fileAdder.Pin = dopin && !hash`,because `hash` is mean not to pin fule.